core/domain/wipeout_service.py

```diff
@@ -841,26 +906,27 @@ def _pseudonymize_models_transactional(
         for metadata_model in metadata_models:
             metadata_model.committer_id = pseudonymized_id
             metadata_model.update_timestamps()
+        datastore_services.put_multi(metadata_models)
 
         commit_log_models = [
             model for model in activity_related_models
             if isinstance(model, commit_log_model_class)]
         for commit_log_model in commit_log_models:
             commit_log_model.user_id = pseudonymized_id
             commit_log_model.update_timestamps()
-        datastore_services.put_multi(metadata_models + commit_log_models)
+        datastore_services.put_multi(commit_log_models)
```

Comment:
Here I splitted put_multi, because previously we are adding two different lists. `metadata_models` is a list of type `base_models.BaseSnapshotMetadataModel` and whereas `commit_log_models` is a list of type `base_models.BaseCommitLogEntryModel`. So, because of this MyPy throws an error and to avoid the error I did this change.

Reply:
@vojtechjelinek Just to check, is this OK? I think we generally want everything to be put at once, so that there isn't a danger of one happening and not the other.

@sahiljoster32 is there a possibility of using union types or similar to satisfy mypy here?

Reply:
Yeah, for `put_multi` we should prioritize doing it only once when possible. Even if some MyPy ignores are needed.

Reply:
> @sahiljoster32 is there a possibility of using union types or similar to satisfy mypy here?

We cannot use Sequence here because we are modifying the list using +, and Sequences are not mutable.

We cannot use List[Union[base_models.BaseSnapshotMetadataModel, base_models.BaseCommitLogEntryModel]] for `commit_log_models` here because attribute `user_id` is only defined in `base_models.BaseCommitLogEntryModel` not in `base_models.BaseSnapshotMetadataModel`. So, MyPy will throw an error for the same.

But we can use something like the below:
```python
combine_func: Callable[[List[Any], List[Any]], List[Any]] = lambda l1, l2: l1+l2

datastore_services.put_multi(combine_func(metadata_models + commit_log_models))
```
this will suppress the error.

And another approach to suppress the error can be:
```python
        all_models: List[base_models.BaseModel] = []
        for model in metadata_models:
            all_models.append(model)
        for model_other in commit_log_models:
            all_models.append(model_other)
        datastore_services.put_multi(all_models)
```
In the second approach we have to use repeated loops so I'm not sure that it is a good approach PTAL and let me know.

For now, i have added an ignore with comment!!

Reply:
Please go with the second approach rather than an ignore. It's fine to have multiple for loops, that's not a problem. It's nested for loops that typically leads to issues.

That said, is it possible to do something like: `for model in metadata_models + commit_log_models` or is that not OK?

Reply:
> Please go with the second approach rather than an ignore

Done!

> is it possible to do something like: for model in metadata_models + commit_log_models or is that not OK?

No, it's not ok, because the main issue is that we are using `+` with two different types of lists.